lib/datadog/grok/src/parse_grok_rules.rs

```diff
@@ -387,6 +390,52 @@ fn resolves_match_function(
             }
             Ok("numberExtStr".to_string())
         }
+        "date" => {
+            return match match_fn.args.as_ref() {
+                Some(args) if !args.is_empty() && args.len() <= 2 => {
+                    if let ast::FunctionArgument::Arg(Value::Bytes(b)) = &args[0] {
+                        let format = String::from_utf8_lossy(b);
+                        let result = date::time_format_to_regex(&format, true)
+                            .map_err(|_e| Error::InvalidFunctionArguments(match_fn.name.clone()))?;
+                        let mut regext_opt = None;
+                        if result.tz_captured {
+                            regext_opt = Some(Regex::new(&result.regex).map_err(|error| {
+                                    error!(message = "Error compiling regex", regex = %result.regex, %error);
+                                    Error::InvalidFunctionArguments(match_fn.name.clone())
+                                })?);
+                        }
+                        let strp_format = date::convert_time_format(&format).map_err(|error| {
+                                error!(message = "Error compiling regex", regex = %result.regex, %error);
+                                Error::InvalidFunctionArguments(match_fn.name.clone())
+                            })?;
+                        let mut target_tz = None;
+                        if args.len() > 1 {
```

Comment:
Could this just be this?

```suggestion
                        if args.len() == 2 {
```